pcuic/theories/README.md

```diff
@@ -2,133 +2,289 @@
-[PCUICSize]: PCUICSize.v
-[PCUICTyping]: PCUICTyping.v
-[PCUICReduction]: PCUICReduction.v
-[PCUICPosition]: PCUICPosition.v
-[PCUICNameless]: PCUICNameless.v
-[PCUICWeakeningEnv]: PCUICWeakeningEnv.v
-[PCUICClosed]: PCUICClosed.v
-[PCUICSigmaCalculus]: PCUICSigmaCalculus.v
-[PCUICWeakening]: PCUICWeakening.v
-[PCUICCumulativity]: PCUICCumulativity.v
-[PCUICEquality]: PCUICEquality.v
-[PCUICSubstitution]: PCUICSubstitution.v
-[PCUICInversion]: PCUICInversion.v
-[PCUICGeneration]: PCUICGeneration.v
-[PCUICParallelReduction]: PCUICParallelReduction.v
-[PCUICParallelReductionConfluence]: PCUICParallelReductionConfluence.v
-[PCUICConfluence]: PCUICConfluence.v
-[PCUICContextConversion]: PCUICContextConversion.v
-[PCUICPrincipality]: PCUICPrincipality.v
-[PCUICUnivSubstitution]: PCUICUnivSubstitution.v
-[PCUICValidity]: PCUICValidity.v
-[PCUICSR]: PCUICSR.v
-[PCUICCSubst]: PCUICCSubst.v
-[PCUICWcbvEval]: PCUICWcbvEval.v
+[PCUICAstUtils]: ./utils/PCUICAstUtils.v
+[PCUICCases]: ./Syntax/PCUICCases.v
+[PCUICInduction]: ./Syntax/PCUICInduction.v
+[PCUICSize]: ./utils/PCUICSize.v
+[PCUICDepth]: ./Syntax/PCUICDepth.v
+[PCUICReflect]: ./Syntax/PCUICReflect.v
+[PCUICContextRelations]: ./Syntax/PCUICContextRelations.v
+[PCUICPosition]: ./Syntax/PUICPosition.v
 
-## Relation with Template-Coq
 
-| File              | Description                                              |
-|-------------------|----------------------------------------------------------|
-| [TemplateToPCUIC] | Translation from Template-Coq syntax to PUIC syntax      |
-| [TemplateToPCUICCorrectness] | Type preservation of the aformentioned translation |
-| [PCUICToTemplate] | Translation from PCUIC syntax to Template-Coq syntax |
-| [PCUICToTemplateCorrectness] | Type preservation of the aformentioned translation |
+## Universes and global environment
+
+| File                | Description                                               |
+|---------------------|-----------------------------------------------------------|
+| [PCUICGlobalEnv]    | Generic properties of global environments
+| [PCUICWfUniverses]  | Definition and properties of a predicate saying that the universe levels of a term are in the global environment
+
+[PCUICGlobalEnv]: ./PCUICGlobalEnv.v
+[PCUICWfUniverses]: ./PCUICWfUniverses.v
+
+
+## Closedness, Renamings and Instantiations
+
+| File                  | Description
+|-----------------------|-------------------------------------------------------
+| [PCUICUnivSubst]      | Substitution of universe variables (for universe polymorphism)
+| [PCUICLiftSubst]      | First commutation properties for lifting and substitution
+| [PCUICSigmaCalculus]  | General theory of renamings
+| [PCUICClosed]         | Properties of the closedness predicate on terms
+| [PCUICOnFreeVars]     | General theory of predicates on (free) variables (generalizing closedness)
+| [PCUICRenameDef]      | Definition of "good" renamings
+| [PCUICInstDef]        | Definition of "good" instantiations
```

Comment:
Would well-behaved be better? Not sure what to use instead, indeed.

Reply:
Maybe yeah. In what sense are they well-behaved?

Reply:
For renaming, every variable on which the predicate `P` holds is sent to another variable, so that the term in the target context corresponding to that variable has the same declaration/definition (up to the right lifting) than the variable in the original context, plus the target context is well-typed.
For substitutions, they send every variable that is defined to either a variable pointing to a definition which is the same in the target context than in the source one, or directly to the body of the definition (with the right lifts again), and moreover the target term has the same type in the target context than the variable had in the source context.

But as you see I’d rather not write this down in the readme, especially since this is exactly the content of the file :)

Reply:
Alright, well-behaved seems good to me.